tests/UnitTest.cpp

```diff
@@ -141,7 +140,7 @@ class IRoom{
     virtual ~IRoom(){}
 };
 
-class ComfortableBed: public virtual IBed{
+class ComfortableBed: public /*virtual*/ IBed{
```

Comment:
I believe these tests shouldn't need `virtual` inheritance anyway. Can you remove them completely instead of commenting it out?